### DISCUSSION TITLE
Refresh session token

### DISCUSSION THREAD

**Discussion by Faiyyaz** [keystonejs/keystone#7907] [Questions]
Hey,

if you take an example of passport authentication, then what we can do there is we can assign two tokens i.e. access token and refresh token. The access token has a shorter life compared to refresh token and then when access token is expired, we pass refresh token to a api and get new access token instead of logging out the user. 

So do we have the same mechanism in keystonejs? Because I cannot find anything in the document related to it.

Thanks!

**Comment by unrevised6419** [comment]
Myself I've been working only with `@keystone-6`.

The Keystone does not use an OAuth/OpenID mechanism.
Instead, it uses a classic cookie approach to track user sessions.

If the user enters the correct username and password, KS will generate a session cookie encrypted with a server secret using [`@hapi/iron`](https://hapi.dev/module/iron/) and save it in the user's browser cookie.

That cookie is valid until it expires. When it expires, the user has to log in again.

If you want to see more technical details, you can look into the stateless session code.

https://github.com/keystonejs/keystone/blob/bd33f0785da2301ec6ff1e7e6ec3a135d6a3cd4c/packages/core/src/session/index.ts#L62-L125

The auth module adds some more code to the session getter.

https://github.com/keystonejs/keystone/blob/bd33f0785da2301ec6ff1e7e6ec3a135d6a3cd4c/packages/auth/src/index.ts#L161-L192

We also used passport and OAuth to authenticate users using third-party providers, but in the end, we still created a user session in KS.

The flow was like this.

- Frontend clicks link that goes to Backend (KS)
- Backend does the redirect to OAuth using the passport
- On the OAuth provider, the user enters credentials
- OAuth provider redirects back to the Backend
- Backend checks if the user exists based on a unique ID from OAuth
  - If the user does not exist, create it based on a unique ID
- Create a Backend KS session (in a way replicate what is [here in KS auth](https://github.com/keystonejs/keystone/blob/bd33f0785da2301ec6ff1e7e6ec3a135d6a3cd4c/packages/auth/src/gql/getBaseAuthSchema.ts#L80-L122)
- Redirect back to Frontend

**Reply by dcousens** [reply]
`@keystone-6/core` is ambivalent to your authentication mechanism; the only thing it needs is a `Session` type.

The answer^ is referring to `@keystone-6/auth` only,  whenyou can easily write your own `SessionStrategy` and UI to match varying requirements.
